include/kernels/transform_reduce.cuh

```diff
@@ -38,6 +38,7 @@ namespace quda {
   template <typename Arg> struct transform_reducer {
     using count_t = decltype(Arg::n_items);
     using reduce_t = decltype(Arg::init_value);
+    using reducer_t = typename Arg::reducer;
```

Comment:
Why is this addition needed here?

Reply:
So I can overload on the reducer type here:
https://github.com/lattice/quda/blob/feature/sycl/include/targets/sycl/reduce_helper.h#L464